string.c

```diff
@@ -6891,6 +6891,7 @@ upcase_single(VALUE str)
 
     while (s < send) {
         unsigned int c = *(unsigned char*)s;
+        const rb_encoding *const enc = 0;
```

Comment:
Rather the call of `rb_enc_isascii` isn't unnecessary?
`'a' <= c && c <= 'z'` implies "isascii" enough, I guess.

Reply:
Sounds reasonable.